source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -23,22 +23,16 @@ function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
 		return;
 	}
 
-	previousFile ??= select('.js-file'); // #5484
-
 	event.preventDefault();
 	event.stopImmediatePropagation();
 
-	const thisFile = event.delegateTarget.closest('.js-file')!;
 	const files = select.all('.js-file');
+	const thisFile = event.delegateTarget.closest('.js-file')!;
+	const isThisFileChecked = isChecked(thisFile);
 
-	const selectedFiles = getItemsBetween<HTMLElement>(files, previousFile, thisFile);
-
-	if (files.indexOf(previousFile!) > files.indexOf(thisFile)) {
-		selectedFiles.reverse();
-	}
```

Comment:
Doesn't really matter in which order the checkboxes are clicked

Reply:
> Doesn't really matter in which order the checkboxes are clicked

This code is related to the call orders of `remember` when `toggleBatch` executed.

You can reproduce the below bug.

The third `batchToggle` is not working as expected in the gif.

![bug1](https://user-images.githubusercontent.com/18283033/163673731-31730c9a-23bd-4084-aa97-a3c2360a0cb9.gif)

This is because `remember` is always called as the array (selectedFiles)'s order.

For example,

```ts
// Expected result: called in order remember(0), remember(1), remember(2)
// This is same with actual result.
// Works as expected.

const selectedFiles = getItemsBetween(files, 0th_item, 3rd_item);

for (const file of selectedFiles) {
	if (isChecked(file) === isThisFileChecked) {
		select('.js-reviewed-checkbox', file)!.click();
	}
}

```

```ts
// Expected result: called in order remember(2), remember(1), remember(0)
// Actual result: called in order remember(0), remember(1), remember(2)
// Because last called `remember` saves the `previousFile`, previousFile will be files[2]. (And it cause the bug.)

const selectedFiles = getItemsBetween(files, 3rd_item, 0th_item);

for (const file of selectedFiles) {
	if (isChecked(file) === isThisFileChecked) {
		select('.js-reviewed-checkbox', file)!.click();
	}
}
```

Reply:
The logic seems faulty:

- if current is checked like `thisFile`, click it

It should be the opposite:

- if it's different, then click it so it matches the current file.

This suggests that `isChecked` might not be working in time so `isThisFileChecked` is actually `wasThisFileCheckedBeforeTheClick`, which is a long condition to keep in mind.

Reply:
> Doesn't really matter in which order the checkboxes are clicked

This should be right, but indeed something is not working in this last commit

Reply:
Ohhhh I finally understood your comment:

> Because last called `remember` saves the `previousFile`

I think we can avoid this by using:

```js
function remember(event: delegate.Event): void {
	if (event.isTrusted) {
		previousFile = event.delegateTarget.closest('.js-file')!;
	}
}
```

This specifically targets manual clicks and excludes everything else